packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -92,7 +93,38 @@ const FormWithRedirect: FC<FormWithRedirectProps> = ({
         [save]
     );
 
-    const formContextValue = useMemo(() => ({ setOnSave }), [setOnSave]);
+    const formContextValue = useMemo<FormContextValue>(
+        () => ({
+            setOnSave,
+            getGroupFields: name => formGroups.current[name] || [],
+            registerGroup: name => {
+                formGroups.current[name] = formGroups.current[name] || [];
+            },
+            unregisterGroup: name => {
+                delete formGroups[name];
+            },
+            registerField: (source, group) => {
+                if (group) {
+                    formGroups.current[group] = formGroups.current[group] || [];
+                    const fields = new Set(formGroups.current[group]);
+                    fields.add(source);
+                    formGroups.current[group] = Array.from(fields);
+                }
```

Comment:
what if group isn't defined? Shouldn't you throw an error?

Reply:
I lazily initialize it here at L108. I found out that inputs were registering themselves before the group anyway.

Reply:
I wasn't clear. My question was: what if `registerField` is called with an empty second argument? But I understand that it's the general case (no group declared) and that in this case there is nothing to do.